src/components/main/files/upload/mod.rs

```diff
@@ -100,3 +114,39 @@ pub fn Upload<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
         ))
     })
 }
+
+
+async fn update_thumbnail(file_storage: Storage, filename_to_save: String) -> Result<String, Error> {
+    let item =  file_storage.root_directory().get_item(&filename_to_save)?;
+    let parts_of_filename: Vec<&str> = filename_to_save.split('.').collect();
+
+    //Since files selected are filtered to be jpg, jpeg, png or svg the last branch is not reachable
+    let mime = match parts_of_filename.last() {
+        Some(m) => {
+            match *m {
+                "png" => IMAGE_PNG.to_string(),
+                "jpg" => IMAGE_JPEG.to_string(),
+                "jpeg" => IMAGE_JPEG.to_string(),
+                "svg" => IMAGE_SVG.to_string(),
+                &_ => "".to_string(),
+            }
+        },
+        None =>  "".to_string(),
+    };
+    
+    let file =  file_storage.get_buffer(&filename_to_save).await?;
+  
+
+    let image = match &file.len() {
```

Comment:
Although images are only be uploaded for now, i would suggest doing something like `if mime.is_empty()` or something like that and inside of the condition to set the thumbnail there. That way if its a generic file it wont be set as apart of the index. I would also suggest looking at a a library to validate the images itself. Something like `image` crate would be better to attempt to decode and validate the image as well as resizing and compressing them to be suitable for thumbnails (unless finding another library suitable for the job too)